### DISCUSSION TITLE
Add option to enable tracing and move logfiles to project drectory

### DISCUSSION THREAD

**Discussion by kpodsiad** [scalameta/metals#3074] [General]
**Use case**
Usually, I've multiple running instances of metals, and in order to see logs from a particular one I have to reload window, so this instance would be the last run one. It annoys me so much 😅

**How it is**
Currently, in order to enable JSON-RPC tracing user has to manually create files ([see manual](https://scalameta.org/metals/docs/contributors/getting-started#json-rpc-trace)). In addition to that, those files are global and only the very last instance of metals can write to them.  

**How it could be**
My proposition is to:
- move the logfiles from home to the current project directory, so logs will be project scoped
- (optionally) add an option to metals, something like "enable json-rpc tracing" switch, and get rid of creating empty files manually.

**Comment by tgodzik** [comment]
I think it's a good idea, though I would maybe add it as a new option? So if you create `~/.metals/lsp.trace.json` it will only log there and not to the global one. What do you think?

**Reply by kpodsiad** [reply]
I couldn't agree with you more!
